### PR TITLE
Login

### PR DESCRIPTION
**Pull request by asherifhegazy**
I just have too add the green background -> I'll get to it later

### CODE REVIEW

File: app/components/login/loginController.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                        toastr.error('Incorrect Username');
+                    }
+                })
+                .catch(function (response) {
+                    console.log(response);
+                });
+        }
+    };
+
+    $scope.usernamePattern = /^(?:[A-Za-z0-9]+)(?:[A-Za-z0-9]*)$/;
```
Comment (dosama1integrant) [inline app/components/login/loginController.js]: We Can define Constant file that carry the patterns defined on the application

File: app/components/login/loginService.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+eShopApp.factory('login', function ($http) {
```
Comment (dosama1integrant) [inline app/components/login/loginService.js]: the "https://localhost:44302/api" part need to be defined as configuration .. we can add app.config.js file to define all configurations on the application as constants and then inject it here to use the defined configuration values from it

**Review by dosama1integrant — CHANGES_REQUESTED** [review]

File: app/components/login/loginController.js (hunk 1 above)
Comment (asherifhegazy) [reply app/components/login/loginController.js]: I made a service for patterns

File: app/components/login/loginService.js (hunk 2 above)
Comment (asherifhegazy) [reply app/components/login/loginService.js]: I created app.config.js as a service so I'll be able to inject it

**Review by dosama1integrant — APPROVED** [review]

**Pull request merged**